packages/send-and-read-emails/backend/java/src/main/java/Server.java

```diff
@@ -0,0 +1,242 @@
+import com.nylas.services.Tunnel;
+import com.nylas.Notification.Delta;
+import io.github.cdimascio.dotenv.Dotenv;
+import io.github.cdimascio.dotenv.DotenvException;
+import spark.utils.StringUtils;
+import utils.User;
+import utils.MockDB;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static spark.Spark.*;
+
+public class Server {
+	public static final Type JSON_MAP = new TypeToken<Map<String, String>>(){}.getType();
+	private static final Gson GSON = new Gson();
+
+	public static void main(String[] args) throws RequestFailedException, IOException, URISyntaxException {
+		Dotenv dotenv = loadEnv();
+
+		// The port the Spark app will run on
+		port(9000);
+
+		// Enable CORS
+		enableCORS();
+
+		// The uri for the frontend
+		String clientUri = dotenv.get("CLIENT_URI", "http://localhost:" + dotenv.get("PORT", "3000"));
+
+		// Initialize an instance of the Nylas SDK using the client credentials
+		NylasApplication application = new NylasClient()
+				.application(dotenv.get("NYLAS_CLIENT_ID"), dotenv.get("NYLAS_CLIENT_SECRET"));
+
+		Tunnel webhookTunnel = new Tunnel.Builder(application, new HandleNotifications()).build();
+		webhookTunnel.connect();
+
+		post("/nylas/generate-auth-url", (request, response) -> {
+			Map<String, String> requestBody = GSON.fromJson(request.body(), JSON_MAP);
+
+			return application.hostedAuthentication()
+					.urlBuilder()
+					.loginHint(requestBody.get("email_address"))
+					.redirectUri(clientUri + requestBody.get("success_url"))
+					.scopes(Scope.EMAIL_READ_ONLY, Scope.EMAIL_MODIFY, Scope.EMAIL_SEND)
+					.buildUrl();
+		});
+
+		/*
+		 * '/nylas/exchange-mailbox-token': This route exchanges an authorization
+		 * code for an access token
+		 * and sends the details of the authenticated user to the client
+		 */
+		post("/nylas/exchange-mailbox-token", (request, response) -> {
+			Map<String, String> requestBody = new Gson().fromJson(request.body(), JSON_MAP);
+			AccessToken accessToken = application.hostedAuthentication()
+					.fetchToken(requestBody.get("token"));
+
+			// Normally store the access token in the DB
+			System.out.println("Access Token was generated for: " + accessToken.getEmailAddress());
+
+			// Replace this mock code with your actual database operations
+			User user = MockDB.createOrUpdateUser(accessToken.getEmailAddress(), accessToken.getAccessToken());
+
+			// Return an authorization object to the user
+			Map<String, String> responsePayload = new HashMap<>();
+			responsePayload.put("id", user.getId());
+			responsePayload.put("emailAddress", user.getEmailAddress());
+			return GSON.toJson(responsePayload);
+		});
+
+		/*
+		 * Before we start our backend, we should whitelist our frontend as a redirect
+		 * URI to ensure the auth completes
+		 */
+		application.addRedirectUri(clientUri);
```

Comment:
move this right under declaring the CLIENT_URI

Reply:
Done!